docs/contracts/StMatic.md

```diff
@@ -406,233 +307,109 @@ Requirements:
+```solidity
+ function requestWithdraw(uint256 _amount)
 ```
-
-### setFee()
-
-Set fee rate to `_feeBasisPoints` basis points. The fees are accrued when oracles report staking results
-
-```sol
-function setFee(uint16 _feeBasisPoints)
-```
-
 #### Parameters
 
 | Name              | Type     | Description                                                    |
 | ----------------- | -------- | -------------------------------------------------------------- |
-| `_feeBasisPoints` | `uint16` | Fee expressed in basis points, 10000 BP corresponding to 100%. |
-
-### setFeeDistribution()
-
-Set fee distribution: `_treasuryFeeBasisPoints` basis points go to the treasury,
-`_insuranceFeeBasisPoints` basis points go to the insurance fund,
-`_operatorsFeeBasisPoints` basis points go to node operators.
-The sum has to be 10 000.
-
-```sol
-function setFeeDistribution(
-  uint16 _treasuryFeeBasisPoints,
-  uint16 _insuranceFeeBasisPoints,
-  uint16 _operatorsFeeBasisPoints
-)
-```
-
-#### Parameters
-
-| Name                       | Type     | Description                                                                            |
-| -------------------------- | -------- | -------------------------------------------------------------------------------------- |
-| `_treasuryFeeBasisPoints`  | `uint16` | Fee for the treasury. Expressed in basis points, 10000 BP corresponding to 100%.       |
-| `_insuranceFeeBasisPoints` | `uint16` | Fee for the insurance fund. Expressed in basis points, 10000 BP corresponding to 100%. |
-| `_operatorsFeeBasisPoints` | `uint16` | Fee for the node operators. Expressed in basis points, 10000 BP corresponding to 100%. |
-
-### setOracle()
-
-Set authorized oracle contract address to `_oracle`
-
-```sol
-function setOracle(address _oracle)
-```
-
-#### Parameters
+| `_amount` | `uint256` | Amount to withdraw. |
 
-| Name      | Type      | Description                |
-| --------- | --------- | -------------------------- |
-| `_oracle` | `address` | Address of oracle contract |
+---
 
-### setTreasury()
-
-Set treasury contract address to `_treasury`. This contract is used to accumulate the protocol treasury fee
-
-```sol
-function setTreasury(address _treasury)
+### delegate()
+Delegates the MATIC deposited into the StMatic contract equally to the validators 
+```solidity
+function delegate() 
 ```
 
-#### Parameters
-
-| Name        | Type      | Description                                        |
-| ----------- | --------- | -------------------------------------------------- |
-| `_treasury` | `address` | Address of contract which accumulates treasury fee |
-
-### setInsuranceFund()
-
-Set insuranceFund contract address to `_insuranceFund`.
-This contract is used to accumulate the protocol insurance fee
+---
 
-```sol
-function setInsuranceFund(address _insuranceFund)
+### claimTokens()
+Allows users claim their tokens from the validators. This requires that the user has a pending withdrawal request
```

Comment:
Allows users to claim their tokens from the validators. This requires user to have an NFT that was minted during the requestWithdraw transaction.